### PR TITLE
Docs: Fix possible values mismatch between document and schema validator

### PR DESCRIPTION
**Pull request by wuxu92**
1. fix possible mismatched values
2. add a possible value list if not present in documents(except timezone type property)
3. remove required/optional for attributes

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
LGTM once merge conflicts are fixed

**Comment by katbyte** [comment]
Also if your written to tool to validate docs against schema might be a good idea to add it in as a linter?

**Comment by wuxu92** [comment]
> Also if your written to tool to validate docs against schema might be a good idea to add it in as a linter?

yes, these doc fix PR was generated by a tool, and I am still working on it. I'll make a PR to azurerm to add it as a linter when it's ready

**Pull request merged**
